pkg/canary/comparator/comparator_test.go

```diff
@@ -218,6 +218,45 @@ func TestEntryNeverReceived(t *testing.T) {
 
 }
 
+// Ensure that if confirmMissing calls pile up and run concurrently, this doesn't cause a panic
+func TestConcurrentConfirmMissing(t *testing.T) {
+	found := []time.Time{
+		time.Unix(0, 0),
+		time.UnixMilli(1),
+		time.UnixMilli(2),
+	}
+	mr := &mockReader{resp: found}
+
+	output := &bytes.Buffer{}
+
+	wait := 30 * time.Millisecond
+	maxWait := 30 * time.Millisecond
+
+	c := NewComparator(output, wait, maxWait, 50*time.Hour, 15*time.Minute, 4*time.Hour, 4*time.Hour, 0, 1*time.Minute, 0, 0, 1, make(chan time.Time), make(chan time.Time), mr, false)
+
+	for _, t := range found {
+		c.missingEntries = append(c.missingEntries, &t)
```

Comment:
All items in the `missingEntries` slice will point to the same `time.Time` object, because `t` inside the range loop is reused for each iteration.

For this test it's irrelevant, but I guess it was not intended.

```suggestion
		tCopy := t
		c.missingEntries = append(c.missingEntries, &tCopy)
```

See https://go.dev/ref/spec#For_statements and https://stackoverflow.com/questions/48826460/using-pointers-in-a-for-loop

Reply:
Ah good catch. This gets me in golang all the time.